lib/src/main/scala/spinal/lib/Utils.scala

```diff
@@ -739,9 +740,16 @@ class CounterUpDown(val stateCount: BigInt, val handleOverflow : Boolean = true)
   val valueNext = UInt(log2Up(stateCount) bit)
   val value = RegNext(valueNext) init(0)
   val mayOverflow = value === stateCount - 1
+
+  assert(value <= stateCount - 1)
+  val mayUnderflow = value === 0
+
   val willOverflowIfInc = mayOverflow && !decrementIt
   val willOverflow = willOverflowIfInc && incrementIt
 
+  val willUnderflowIfDec = mayUnderflow && !incrementIt
```

Comment:
If this is not only for verification refactor, please open a PR individually.

Reply:
Will do